src/Microsoft.AspNet.Mvc.Core/ModelBinding/BodyModelBinder.cs

```diff
@@ -64,9 +64,9 @@ public BodyModelBinder()
                     // system to skip other model binders and never to fall back.
                     return new ModelBindingResult(modelBindingKey);
                 }
-
-                var valueProviderResult = new ValueProviderResult(rawValue: model);
-                bindingContext.ModelState.SetModelValue(modelBindingKey, valueProviderResult);
+                
+                // We don't have anything useful to put in model state.
```

Comment:
> Huh? We changed this code not that long ago to explicitly get the model into the ModelStateDictionary. 

Yes, and this is an intentional change back from that. Storing arbitrary objects in `RawValue` adds more odd and barely tested scenarios for the conversion code + HTML helpers have to deal with. This change gets on track with the API supporting what `RawValue` was intended to be used for.

We don't gain anything from storing deserialized models in `RawValue` - it's just something we did for consistency. In the future, if we find a need to store the bound model value, we should create a new slot in MSD and do that instead of overloading `RawValue`.